File: apps/ios/Pebbles/Features/Glyph/Models/GlyphStroke.swift
import Foundation

/// One stroke within a carved glyph. Mirrors the web `MarkStroke` shape so
/// glyphs are interoperable between web and iOS.
///
/// - `d`: SVG path string ("M x,y L x,y …" or with quadratic Beziers for smoothed strokes).
/// - `width`: stroke width in the glyph's 200x200 coordinate space. iOS-carved
///   strokes always use 6 per the issue constraint (no user-facing slider).
struct GlyphStroke: Codable, Hashable {
    let d: String
    let width: Double
}

File: apps/android/app/src/main/kotlin/app/pbbls/android/features/glyph/models/GlyphStroke.kt
package app.pbbls.android.features.glyph.models

import kotlinx.serialization.Serializable

/**
 * One stroke within a carved glyph — mirrors iOS `GlyphStroke.swift` and the
 * web `MarkStroke` shape so glyphs are interoperable across surfaces.
 *
 * [d] is an SVG path string (`M x,y L x,y …`, or quadratic Béziers for
 * smoothed strokes). [width] is the stroke width in the glyph's 200×200
 * coordinate space.
 */
@Serializable
data class GlyphStroke(
    val d: String,
    val width: Double,
)
